### PR TITLE
Make project_metadata mandatory in scan_request

### PR DESCRIPTION
**Pull request by aryx**
Step1 towards getting rid of meta.

test plan:
see related PR in semgrep-pro

- [x] I ran `make setup && make` to update the generated code after editing a `.atd` file (TODO: have a CI check)
- [x] I made sure we're still backward compatible with old versions of the CLI.
      For example, the Semgrep backend need to still be able to *consume* data generated
	  by Semgrep 1.50.0.
      See https://atd.readthedocs.io/en/latest/atdgen-tutorial.html#smooth-protocol-upgrades

### CODE REVIEW

**Comment by chmccreery** [comment]
> For example, the Semgrep backend need to still be able to consume data generated
by Semgrep 1.17.0.

Probably time to update this text to say 1.50.0 😄

**Review by chmccreery — APPROVED** [review]

**Comment by aryx** [comment]
Note that the backward compatiblity failures are expected. We are breaking backward compatibility on purpose.
We're supporting only versions since 1.50.0 and we know we've been producing those fields since that version for sure (we in fact started in 1.43.0 so we're good).

File: semgrep_output_v1.atd [hunk 1]
```diff
@@ -1444,9 +1444,9 @@ type scan_request = {
     (* until 1.43ish, was 'meta: project_metadata;' before *)
     meta: raw_json;
     (* from 1.43 *)
-    ?project_metadata: project_metadata option; (* replacing meta *)
-    ?project_config: ci_config_from_repo option;
-    ?scan_metadata: scan_metadata option;
+    project_metadata: project_metadata; (* replacing meta *)
+    project_config: ci_config_from_repo;
+    scan_metadata: scan_metadata;
 }
```
[the remark this answers is not in the document]
Comment (aryx) [reply semgrep_output_v1.atd]: I've put back project_config as an option. There might not be a .semgrepconfig.yml in the repo so I think it's more logical to have an option here.

**Comment by aryx** [comment]
new version in https://github.com/semgrep/semgrep-interfaces/pull/324

**Pull request closed without merging**